bayesian/utils.py

```diff
@@ -198,23 +198,22 @@ def generate_recommendation(data, package, version):
                     reco['recommendation'] = {}
                     return {"result": reco}
 
-        # check if latest version exists or current version is latest version
-        if not latest_version or latest_version == '' or version == latest_version:
-            if message != '':
-                reco['recommendation']['message'] = message
-            return {"result": reco}
-        # check if latest version has lower CVEs or no CVEs than current version
         for records in data:
             ver = records['version']
-            if latest_version == ver.get('version', [''])[0]:
-                if ver.get('cve_ids', [''])[0] != '':
-                    for cve in ver.get('cve_ids'):
-                        cvss = float(cve.split(':')[1])
-                        if cvss >= max_cvss:
-                            break
-                message += '\n It is recommended to use Version - ' + latest_version
-                reco['recommendation']['change_to'] = latest_version
-                reco['recommendation']['message'] = message
+            graph_version = ver.get('version', [''])[0]
+            graph_ver = convert_version_to_proper_semantic(graph_version)
+
+            # Check for next best higher version than input version without any CVE's
+            if (ver.get('cve_ids', [''])[0] == '' or 'cve_ids' not in ver) \
+                    and sv.Version(graph_ver) > sv.Version(ip_ver):
+                if not higher_version:
+                    higher_version = graph_ver
+                if sv.Version(higher_version) > sv.Version(graph_ver):
+                    higher_version = graph_ver
```

Comment:
Shouldn't this be:
```python
if sv.Version(graph_ver) > sv.Version(higher_version):
    higher_version = graph_ver

```

Reply:
Actually we want the closest upgraded version to a user's version. If user's version is 1.0.0 and we have 1.0.1 and 1.0.4 without any CVE, we would like to recommend a closer version to the user's which would be 1.0.1 in this case.

Reply:
makes sense :+1: